gn2/wqflask/views.py

```diff
@@ -268,11 +268,47 @@ def gsearchtable():
 
     return flask.jsonify(current_page)
 
+
 def clean_xapian_query(query: str) -> str:
-    """ Remove filler words in xapian query
-    TODO: FIXME
     """
-    return query
+    Clean and optimize a Xapian query string by removing filler words,
+    and ensuring the query is tailored for optimal results from Fahamu.
+
+    Args:
+        query (str): The original Xapian query string.
+
+    Returns:
+        str: The cleaned and optimized query string.
+    """
+    xapian_prefixes = {
+        "author",
+        "species",
+        "group",
+        "tissue",
+        "dataset",
+        "symbol",
+        "description",
+        "rif",
+        "wiki",
+    }
+    xapian_operators = {"AND", "NOT", "OR", "XOR", "NEAR", "ADJ"}
+    range_prefixes = {"mean", "peak", "position", "peakmb", "additive", "year"}
+    query_context = ["genes"]
+    cleaned_query_parts = []
+    for token in query.split():
+        if token in xapian_operators or any(
+            prefix in token for prefix in range_prefixes if ".." in token
```

Comment:
What conditions are we handling here? I can only think of `mean:8..` and `Hs:chr4:9930021`. These can be handled by:
```
if prefix in range_prefixes:
  continue
```
I'm not sure the `".." in token` is needed

Reply:
These conditions are handled where token is 

```
XAPIAN OPERATOR 
RANGE 
```
More on this:
From  the docs this is the range syntax is word:
 https://xapian.org/docs/queryparser.html. 
This means that they are only considered ranges if they include the range operator (`..`) as part of their token. For example, `mean:5...7` should be rejected, whereas `mean` should be accepted as a word. 
Ideally this check should be  token has : and also .. and strip the word too
I will improve on this for clarity.

Reply:
ah, I see. Thanks for the explanation. Then can we change this to:

```
prefix, _, suffix = token.partition(":") # line already defined below
if ".." in suffix and prefix in range_prefixes:
  continue
```